drivers/include/periph/timer.h

```diff
@@ -146,6 +148,33 @@ int timer_clear(tim_t dev, int channel);
  */
 unsigned int timer_read(tim_t dev);
 
+/**
+ * @brief Get the timers maximal counter value.
+ *
+ * @param[in] dev           the timer to get the max value for
+ *
+ * @return                  the timers max value
+ * @return                  0 on error
+ */
+unsigned int timer_max(tim_t dev);
```

Comment:
Is it theoretically possible that a platform has only 16-bit unsigned  int, but the maximum timer value has a width of 32 bits?

Reply:
I hope not, because then our API is not capable of reflecting that 😬

Reply:
because `timer_read` also only return `unsigned int`

Reply:
I have no platform in mind, it was just a thought what if.